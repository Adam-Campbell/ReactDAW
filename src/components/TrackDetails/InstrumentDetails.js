import React from 'react';
import PropTypes from 'prop-types';
import { synthTypes } from '../../constants';

const InstrumentDetails = props => (
    <div className="track-details__instrument-info-container">
        <h2 className="track-details__heading">Instrument</h2>
        <label 
            htmlFor="instrument-select" 
            className="track-details__instrument-select-label"
        >Current instrument</label>
        <select 
            id="instrument-select" 
            className="track-details__instrument-select"
            value={props.instrumentType}
            onChange={props.handleChange}
        >
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.default}
            >Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.am}
            >AM Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.fm}
            >FM Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.duo}
            >Duo Synth</option>
            <option 
                className="track-details__instrument-select-option" 
                value={synthTypes.mono}
            >Mono</option>
        </select>
        <button 
            className="track-details__instrument-settings-button"
            onClick={props.handleOpen}
        >Edit Settings</button>
    </div>
);

InstrumentDetails.propTypes = {
    instrumentType: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleOpen: PropTypes.func.isRequired
};

export default InstrumentDetails;