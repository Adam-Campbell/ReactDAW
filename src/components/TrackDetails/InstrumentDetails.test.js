import React from 'react';
import { shallow } from 'enzyme';
import InstrumentDetails from './InstrumentDetails';

test('renders correctly', () => {
    const component = shallow(
        <InstrumentDetails 
            instrumentType="synth"
            handleChange={jest.fn()}
            handleOpen={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});