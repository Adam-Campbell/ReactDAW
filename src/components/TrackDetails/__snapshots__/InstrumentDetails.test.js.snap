// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="track-details__instrument-info-container"
>
  <h2
    className="track-details__heading"
  >
    Instrument
  </h2>
  <label
    className="track-details__instrument-select-label"
    htmlFor="instrument-select"
  >
    Current instrument
  </label>
  <select
    className="track-details__instrument-select"
    id="instrument-select"
    onChange={[MockFunction]}
    value="synth"
  >
    <option
      className="track-details__instrument-select-option"
      value="synth"
    >
      Synth
    </option>
    <option
      className="track-details__instrument-select-option"
      value="amSynth"
    >
      AM Synth
    </option>
    <option
      className="track-details__instrument-select-option"
      value="fmSynth"
    >
      FM Synth
    </option>
    <option
      className="track-details__instrument-select-option"
      value="duoSynth"
    >
      Duo Synth
    </option>
    <option
      className="track-details__instrument-select-option"
      value="monoSynth"
    >
      Mono
    </option>
    <option
      className="track-details__instrument-select-option"
      value="drumKit"
    >
      Drum Kit
    </option>
  </select>
  <button
    className="track-details__instrument-settings-button"
    onClick={[MockFunction]}
  >
    Edit Settings
  </button>
</div>
`;
